### ISSUE TITLE
Trap error when trying to find indices on stretched grid

### ISSUE THREAD

**Issue by tclune** [GEOS-ESM/MAPL#2643]
The latest horzIJindex procedure (see #2636) presumably fails when the grid is stretched.

We need to detect that and fail with an informative error message.

**Comment by weiyuan-jiang** [comment]
I will add a print statement next time when I get chance

**Comment by mathomp4** [comment]
Hmm. @weiyuan-jiang has this been fixed? I want to say yes...

**Comment by lizziel** [comment]
Yuanjian Zhang (@yuanjianz, WashU) brought up this issue with regards to stretched grid GCHP runs. Please reopen as we need this to be fixed. See also https://github.com/geoschem/GCHP/issues/549.

**Comment by lizziel** [comment]
By fixed, I mean we need there to be no need for an error trap. I can make a new feature request for this.

**Comment by tclune** [comment]
@weiyuan-jiang Please comment on the status of this issue.

**Comment by tclune** [comment]
I'm fairly certain that we produce correct indices now.   The code applies the inverse stretch and then bins the point on a non-stretched cube.    Roundoff issues aside.

**Comment by weiyuan-jiang** [comment]
I need more information to figure out what happens. Is this stretched grid created by MAPL ? How big is the difference between accurate lons and the lons of the grid?

**Comment by yuanjianz** [comment]
@weiyuan-jiang yes, this stretched grid is created by MAPL. I am not sure about the exact difference, but it is definitely larger than the tolerance so it aborts. I tested over C48 sf 2d1 and target lat of 44.0 and lon of 263.0.

**Comment by weiyuan-jiang** [comment]
@yuanjianz Is it easy to setup a case for me to run?

**Comment by yuanjianz** [comment]
@weiyuan-jiang I think you don't necessary need to set up a GCHP simulation. All you need to do is create a MAPL grid with the stretching parameter I listed above and call MAPL_GetHorzIJIndex on it. The lat and lon needed for the inquiry are volcano emissions which can be found in 

[so2_volcanic_emissions_Carns.20190701.rc.txt](https://github.com/user-attachments/files/27645639/so2_volcanic_emissions_Carns.20190701.rc.txt)

Or you can try to randomly select lats and lons, and I think many of them will trigger this trap.

**Comment by weiyuan-jiang** [comment]
What is the stretch factor?

**Comment by yuanjianz** [comment]
> What is the stretch factor?

stretch factor of 2.1, target lat 44.0, target lon 263.0

**Comment by weiyuan-jiang** [comment]
In stretch grid, the corner lats lons are calculated with r4, but here the accurate ones are calculated with r8.  So we have two choices: (1) relax the comparison standard ( from 2.0tolerance to 10.0 tolerance or even more  (2) set the corner lats lans with r8. @yuanjianz 
@tclune which choice do you prefer?

**Comment by weiyuan-jiang** [comment]
@yuanjianz , would you please try this change? https://github.com/GEOS-ESM/MAPL/blob/ef768dcf2f2034fae07ad927ba50d7ae29deb3f4/base/Base/Base_Base_implementation.F90#L2891

**Comment by yuanjianz** [comment]
@weiyuan-jiang changing to 10*tolerance fix the abort as expected.

**Comment by lizziel** [comment]
Could this fix go into the next release of MAPL this week @mathomp4?

**Comment by mathomp4** [comment]
Sure. @weiyuan-jiang Can you make a PR to `develop` with this?

**Comment by mathomp4** [comment]
Ah @weiyuan-jiang already did that in https://github.com/GEOS-ESM/MAPL/pull/4870

So, yes, @lizziel it should be in the next release
